client/extensions/woocommerce/app/settings/shipping/shipping-zone/index.js

```diff
@@ -72,7 +77,22 @@ class Shipping extends Component {
 	}
 
 	onSave() {
-		const { siteId, zone, locations, translate, actions } = this.props;
+		const { siteId, zone, locations, translate, actions, areLocationsValid, areMethodsValid } = this.props;
+		let hasErrors = false;
+		if ( ! areLocationsValid ) {
+			actions.errorNotice( translate( 'Add at least one location to this zone' ), { duration: 4000 } );
+			hasErrors = true;
+		}
+
+		if ( ! areMethodsValid ) {
+			actions.errorNotice( translate( 'Add shipping methods to this zone' ), { duration: 4000 } );
```

Comment:
All there checks would be cleaner in the `action-list` creation method (`WOOCOMMERCE_SHIPPING_ZONE_ACTION_LIST_CREATE`), this feels like too much logic to have in a component.